mater/lib/src/stores/filestore.rs

```diff
@@ -76,18 +84,111 @@ where
     let placeholder_header_v1 = CarV1Header::default();
     position += writer.write_v1_header(&placeholder_header_v1).await?;
 
-    let mut root = None;
+    let mut root: Option<Cid> = None;
     let mut entries = vec![];
+
+    // Collect blocks and track final root
     while let Some(node) = nodes.next().await {
         let (node_cid, node_bytes) = node?;
-        let digest = node_cid.hash().digest().to_owned();
+        let digest = node_cid.hash().digest().to_vec();
+
+        // Index entry for the block
         let entry = IndexEntry::new(digest, (position - car_v1_start) as u64);
         entries.push(entry);
+
+        // Write the block
         position += writer.write_block(&node_cid, &node_bytes).await?;
```

Comment:
A slice's `to_owned` implementation is a `to_vec`, but the function call was selected on purpose to *communicate intent*. The comments after repeat what the struct (`IndexEntry`) and function `write_block` already say
```suggestion
        let digest = node_cid.hash().digest().to_owned();
        let entry = IndexEntry::new(digest, (position - car_v1_start) as u64);
        entries.push(entry);
        position += writer.write_block(&node_cid, &node_bytes).await?;
```